src/MultiOwnerLightAccount.sol

```diff
@@ -0,0 +1,336 @@
+        uint256 length = ownersToAdd.length;
+        for (uint256 i = 0; i < length; ++i) {
+            address ownerToAdd = ownersToAdd[i];
+            if (
+                ownerToAdd == address(0) || ownerToAdd == address(this)
+                    || !_storage.owners.tryAdd(ownerToAdd.toSetValue())
+            ) {
+                revert InvalidOwner(ownerToAdd);
+            }
+        }
+    }
+
+    function _removeOwnersOrRevert(address[] memory ownersToRemove) internal {
+        LightAccountStorage storage _storage = _getStorage();
+        uint256 length = ownersToRemove.length;
+        for (uint256 i = 0; i < length; ++i) {
+            if (!_storage.owners.tryRemove(ownersToRemove[i].toSetValue())) {
+                revert OwnerDoesNotExist(ownersToRemove[i]);
+            }
+        }
+    }
+
+    /// @dev Implement template method of BaseAccount.
+    /// Uses a modified version of `SignatureChecker.isValidSignatureNow` in which the digest is wrapped with an
+    /// "Ethereum Signed Message" envelope for the EOA-owner case but not in the ERC-1271 contract-owner case.
+    function _validateSignature(PackedUserOperation calldata userOp, bytes32 userOpHash)
+        internal
+        virtual
+        override
+        returns (uint256 validationData)
+    {
+        bytes32 signedHash = userOpHash.toEthSignedMessageHash();
+        bytes memory signature = userOp.signature;
+        (address recovered, ECDSA.RecoverError error,) = signedHash.tryRecover(signature);
+        if (error == ECDSA.RecoverError.NoError && _getStorage().owners.contains(recovered.toSetValue())) {
+            return 0;
+        }
+        if (_isValidERC1271SignatureNow(userOpHash, signature)) {
+            return 0;
+        }
+        return SIG_VALIDATION_FAILED;
+    }
+
+    function _isValidERC1271SignatureNow(bytes32 digest, bytes memory signature) internal view returns (bool) {
+        LightAccountStorage storage _storage = _getStorage();
+        address[] memory owners_ = _storage.owners.getAll().toAddressArray();
+        uint256 length = owners_.length;
+        for (uint256 i = 0; i < length; ++i) {
+            if (SignatureChecker.isValidERC1271SignatureNow(owners_[i], digest, signature)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /// @dev Revert if the caller is not one of the owners or the account itself (when redirected through `execute`).
+    function _onlyOwners() internal view {
+        if (msg.sender != address(this) && !_getStorage().owners.contains(msg.sender.toSetValue())) {
+            revert NotAuthorized(msg.sender);
+        }
+    }
+
+    /// @dev Require that the call is from the entry point or an owner.
+    function _requireFromEntryPointOrOwner() internal view {
+        if (msg.sender != address(entryPoint()) && !_getStorage().owners.contains(msg.sender.toSetValue())) {
+            revert NotAuthorized(msg.sender);
+        }
+    }
```

Comment:
Would it make sense to merge these two modifiers? There are some workflows where needing the call to be wrapped in `execute` is just an added step, like `updateOwners`.